library/src/window_impl.cxx

```diff
@@ -147,45 +171,155 @@ class window_impl::internals
+    }
+    this->Internals->RenWin = vtkSmartPointer<vtkEGLRenderWindow>::New();
 #ifdef __ANDROID__
     // Since F3D_MODULE_EXTERNAL_RENDERING is not supported on Android yet, we need to call
     // this workaround. It makes vtkEGLRenderWindow external if WindowInfo is not nullptr.
     this->Internals->RenWin->SetWindowInfo("jni");
+#endif
+#else
+    throw engine::no_window_exception("Window type is EGL but VTK EGL support is not enabled");
+#endif
+  }
+  else if (type == Type::OSMESA)
+  {
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240914)
+    this->Internals->RenWin = vtkSmartPointer<vtkOSOpenGLRenderWindow>::New();
+#else
+    throw engine::no_window_exception(
+      "Window type is OSMesa but VTK OSMesa support is not enabled");
+#endif
+  }
+  else if (type == Type::GLX)
+  {
+#ifdef VTK_USE_X
+    this->Internals->RenWin = vtkSmartPointer<vtkXOpenGLRenderWindow>::New();
+#else
+    throw engine::no_window_exception("Window type is GLX but VTK GLX support is not enabled");
+#endif
+  }
+  else if (type == Type::WGL)
+  {
+#ifdef _WIN32
+    this->Internals->RenWin = vtkSmartPointer<vtkWin32OpenGLRenderWindow>::New();
+#else
+    throw engine::no_window_exception("Window type is WGL but it is supported on Windows only");
 #endif
   }
+  else if (type == Type::COCOA)
+  {
+#ifdef __APPLE__
+    this->Internals->RenWin = vtkSmartPointer<vtkCocoaRenderWindow>::New();
+#else
+    throw engine::no_window_exception("Window type is Cocoa but it is supported on macOS only");
+#endif
+  }
+  else if (type == Type::AUTO)
+  {
+    this->Internals->RenWin = vtkSmartPointer<vtkRenderWindow>::New();
+
+    if (this->Internals->RenWin->IsA("vtkOSOpenGLRenderWindow"))
+    {
+      this->Internals->WindowType = Type::OSMESA;
+    }
+#ifdef VTK_USE_X
+    else if (this->Internals->RenWin->IsA("vtkXOpenGLRenderWindow"))
+    {
+      this->Internals->WindowType = Type::GLX;
+    }
+#endif
+#ifdef _WIN32
+    else if (this->Internals->RenWin->IsA("vtkWin32OpenGLRenderWindow"))
+    {
+      this->Internals->WindowType = Type::WGL;
+    }
+#endif
+#ifdef __APPLE__
+    else if (this->Internals->RenWin->IsA("vtkCocoaRenderWindow"))
+    {
+      this->Internals->WindowType = Type::COCOA;
+    }
+#endif
+#ifdef VTK_OPENGL_HAS_EGL
+    else if (this->Internals->RenWin->IsA("vtkEGLRenderWindow"))
+    {
+      this->Internals->WindowType = Type::EGL;
+    }
+#endif
+#ifdef __EMSCRIPTEN__
+    else if (this->Internals->RenWin->IsA("vtkWebAssemblyOpenGLRenderWindow"))
+    {
+      this->Internals->WindowType = Type::WASM;
+    }
+#endif
+    else
+    {
+      this->Internals->WindowType = Type::UNKNOWN;
+    }
+  }
+  else
+  {
+    throw engine::no_window_exception("Cannot create this window type.");
+  }
+
 #if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240606)
   this->Internals->RenWin->EnableTranslucentSurfaceOn();
 #endif
+  this->Internals->RenWin->SetMultiSamples(0); // Disable hardware antialiasing
```

Comment:
why does this matter ?

Reply:
It was already there, I just moved this line. VTK does not work well with multi-sampling, and I'd rather continue using FXAA when we need anti-aliasing.